modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/httpsession/GetSessionParam.java

```diff
@@ -0,0 +1,124 @@
+        value = "Create a new session or not") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "Session",
+        value = "HTTP session struct") })
+public class GetSessionParam extends AbstractNativeFunction {
+
+    private static final Logger logger = LoggerFactory.getLogger(GetSessionParam.class);
+
+    @Override
+    public BValue[] execute(Context context) {
+        try {
+            CarbonMessage carbonMessage = ((BMessage) getRefArgument(context, 0)).value();
+            Boolean create = getBooleanArgument(context, 0);
+            String cookieHeader = carbonMessage.getHeader(COOKIE_HEADER);
+            String path = (String) carbonMessage.getProperty(BASE_PATH);
+            Session session = context.getCurrentSession();
+
+            if (cookieHeader != null) {
+                try {
+                    String sessionId = Arrays.stream(cookieHeader.split(";"))
+                                        .filter(cookie -> cookie.startsWith(SESSION_ID))
+                                        .findFirst().get().substring(SESSION_ID.length());
+                    //return value from cached session
+                    if (session != null && (sessionId.equals(session.getId()))) {
+                        session = session.setAccessed();
+                        return new BValue[]{GetSession.createSessionStruct(context, session)};
+                    }
+                    session = context.getSessionManager().getHTTPSession(sessionId);
+                } catch (NoSuchElementException e) {
+                    //ignore throwable
+                    logger.info("Failed to get session: Incorrect Session cookie");
+                }
+                if (session != null) {
+                    //path Validity check
+                    if (session.getPath().equals(path)) {
+                        session.setNew(false);
+                        session.setAccessed();
+                    } else {
+                        throw new BallerinaException("Failed to get session: " + path + " is not an allowed path");
+                    }
+                } else {
+                    if (create) {
+                        session = context.getSessionManager().createHTTPSession(path);
+                    } else {
+                        //Return null if create is false.
+                        logger.info("Session is not created");
+                        return new BValue[]{};
+                    }
+                }
+            } else if (create) {
+                session = context.getSessionManager().createHTTPSession(path);
+            }
+            carbonMessage.removeHeader(COOKIE_HEADER);
+            if (session != null) {
+                context.setCurrentSession(session);
+                return new BValue[]{GetSession.createSessionStruct(context, session)};
+            }
+            return new BValue[]{};
+        } catch (NullPointerException e) {
```

Comment:
Catching a NullPointerException is wrong. Check null and throw appropriate exception.